App build failure by circular dependency between kapt and ksp

**Please complete the following information:**
* Build failure
 
**Describe the Bug:**
```
$ git clone https://github.com/skydoves/Pokedex

$ cd Pokedex

$ ./gradlew :app:aR

FAILURE: Build failed with an exception.

* What went wrong:
Circular dependency between the following tasks:
:app:kaptGenerateStubsReleaseKotlin
\--- :app:kspReleaseKotlin
     \--- :app:kaptReleaseKotlin
          \--- :app:kaptGenerateStubsReleaseKotlin (*)

(*) - details omitted (listed previously)

```

Here is gradle scan result : https://scans.gradle.com/s/2xsbxrhuos4b4

* commit : 9b51574e
* related issue: https://github.com/google/ksp/issues/1772


**Expected Behavior:**

--- a/.gitignore
+++ b/.gitignore
@@ -58,3 +58,8 @@ google-services.json
 freeline.py
 freeline/
 freeline_project_description.json
+
+# Eclipse
+.project
+.settings
+.classpath

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -28,12 +28,10 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex"
 
   defaultConfig {
     applicationId = "com.skydoves.pokedex"
-    minSdk = Configuration.minSdk
     targetSdk = Configuration.targetSdk
     versionCode = Configuration.versionCode
     versionName = Configuration.versionName

--- a/benchmark/build.gradle.kts
+++ b/benchmark/build.gradle.kts
@@ -21,7 +21,6 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.benchmark"
 
   compileOptions {
@@ -34,8 +33,7 @@ android {
   }
 
   defaultConfig {
-    minSdk = Configuration.minSdk
-    targetSdk = Configuration.compileSdk
+    targetSdk = Configuration.targetSdk
     testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
   }
 

--- a/buildSrc/src/main/kotlin/com/skydoves/pokedex/Configuration.kt
+++ b/buildSrc/src/main/kotlin/com/skydoves/pokedex/Configuration.kt
@@ -1,9 +1,7 @@
 package com.skydoves.pokedex
 
 object Configuration {
-  const val compileSdk = 34
   const val targetSdk = 34
-  const val minSdk = 23
   const val majorVersion = 1
   const val minorVersion = 1
   const val patchVersion = 2

--- a/core-data/build.gradle.kts
+++ b/core-data/build.gradle.kts
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import com.skydoves.pokedex.Configuration
 
 plugins {
   alias(libs.plugins.android.library)
@@ -22,12 +21,7 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.core.data"
-
-  defaultConfig {
-    minSdk = Configuration.minSdk
-  }
 
   compileOptions {
     sourceCompatibility = JavaVersion.VERSION_17

--- a/core-database/build.gradle.kts
+++ b/core-database/build.gradle.kts
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import com.skydoves.pokedex.Configuration
 
 plugins {
   alias(libs.plugins.android.library)
@@ -22,11 +21,9 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.core.database"
 
   defaultConfig {
-    minSdk = Configuration.minSdk
     // The schemas directory contains a schema file for each version of the Room database.
     // This is required to enable Room auto migrations.
     // See https://developer.android.com/reference/kotlin/androidx/room/AutoMigration.

--- a/core-model/build.gradle.kts
+++ b/core-model/build.gradle.kts
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import com.skydoves.pokedex.Configuration
 
 plugins {
   alias(libs.plugins.android.library)
@@ -23,12 +22,7 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.core.model"
-
-  defaultConfig {
-    minSdk = Configuration.minSdk
-  }
 
   compileOptions {
     sourceCompatibility = JavaVersion.VERSION_17

--- a/core-network/build.gradle.kts
+++ b/core-network/build.gradle.kts
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import com.skydoves.pokedex.Configuration
 
 plugins {
   alias(libs.plugins.android.library)
@@ -22,12 +21,7 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.core.network"
-
-  defaultConfig {
-    minSdk = Configuration.minSdk
-  }
 
   compileOptions {
     sourceCompatibility = JavaVersion.VERSION_17

--- a/core-test/build.gradle.kts
+++ b/core-test/build.gradle.kts
@@ -13,7 +13,6 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import com.skydoves.pokedex.Configuration
 
 plugins {
   alias(libs.plugins.android.library)
@@ -21,12 +20,7 @@ plugins {
 }
 
 android {
-  compileSdk = Configuration.compileSdk
   namespace = "com.skydoves.pokedex.core.test"
-
-  defaultConfig {
-    minSdk = Configuration.minSdk
-  }
 
   compileOptions {
     sourceCompatibility = JavaVersion.VERSION_17

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,7 +1,7 @@
 [versions]
 agp = "8.2.2"
 kotlin = "1.9.22"
-ksp = "1.9.22-1.0.18"
+ksp = "1.9.22-1.0.17"
 jvmTarget = "17"
 material = "1.11.0"
 androidxActivity = "1.7.0"

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -58,6 +58,16 @@ pluginManagement {
     }
   }
 }
+
+plugins {
+  id("com.android.settings") version "8.3.0"
+}
+
+android {
+  minSdk = 23
+  compileSdk = 34
+}
+
 dependencyResolutionManagement {
   repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
   repositories {

